js/ng.js

```diff
@@ -463,4 +544,92 @@ async function deleteGloss(id=glossHashID) {
         console.log(err)
     })
 
-}
\ No newline at end of file
+}
+
+async function queryBibliographicCitations(glossId) {
+    const query = {
+        "@type": "BibliographicCitation",
+        "references": glossId, 
+        "__rerum.generatedBy": __constants.generator
+    };
+
+    try {
+        const res = await fetch(`${__constants.tiny+"/query"}`, {
+            method: 'POST',
+            mode: 'cors',
+            headers: {
+                "Content-Type": "application/json;charset=utf-8"
+            },
+            body: JSON.stringify(query)
+        }).then(response => response.json());
+
+        return res
+    } catch (err) {
+        console.error(err);
+    }
+}
+
+async function addBibliographicCitationToGloss(bibliographicCitation, glossId) {
+    try {
+        if (typeof bibliographicCitation !== 'string' || !bibliographicCitation.trim()) {
+            const invalidInputEvent = new CustomEvent("Invalid bibliographic citation input: must be a non-empty string.")
+            globalFeedbackBlip(invalidInputEvent, 'Invalid bibliographic citation input: must be a non-empty string.', false)
+            return;
+        }
+
+        const cleanCitation = bibliographicCitation.trim();
+
+        const query = {
+            "@type": "BibliographicCitation",
+            "references": glossId,
```

Comment:
Careful here. We use references elsewhere but track it as an Array, even when only a single ID.